### DISCUSSION TITLE
Assign the register text value to radio button value

### DISCUSSION THREAD

**Discussion by jjdechavez** [react-hook-form/react-hook-form#6135] [Q&A]
**Expectation**
Whenever I append a Radio button and Text field on Nested Fields. I want to assign the value of radio button based on text field value.

**To Reproduce**
1. Press the Append Multiple Choice Button.
2. Input the value on the text field.
3. Add a new Option by clicking the Append Option Button.
4. Input the value on the new text field.
5. Submit it by clicking Submit Query Button.
6. Check the `template.questions[0].answer` from `console.log()`.

Expected Result:
`template.questions[0].answer` must have value based on selected radio button.

Actual Result:
`template.questions[0].answer` is empty string.

**Findings**
Whenever I append a Option(Radio button and text field) and assign the value to the text field it doesn't update the value of the radio button. So, when I created a new Option the previous Option updates the radio button value. I'm thinking if it's render issue that why the radio button doesn't update the value.

**CodeSandBox**
https://codesandbox.io/s/summer-cherry-jmyc2?file=/src/components/Form.js

Let me know if you need more details about this. Thank you!

**Comment by bluebill1049** [comment]
use `Controller` and build a controlled input group instead.
